src/lib.rs

```diff
@@ -260,61 +294,79 @@ impl Watch {
     }
 }
 
+#[non_exhaustive]
 #[derive(Debug, StructOpt)]
 pub struct DevServer {
     #[structopt(long, default_value = "127.0.0.1")]
-    ip: IpAddr,
+    pub ip: IpAddr,
     #[structopt(long, default_value = "8000")]
-    port: u16,
+    pub port: u16,
 
+    #[structopt(skip)]
+    pub served_path: Option<PathBuf>,
+    #[structopt(skip)]
+    pub release: bool,
     #[structopt(flatten)]
-    watch: Watch,
+    pub watch: Watch,
 }
 
 impl DevServer {
-    pub fn serve(&self, build_dir_path: impl AsRef<Path>) -> Result<()> {
-        let address = SocketAddr::new(self.ip, self.port);
-        let listener = TcpListener::bind(&address).context("cannot bind to the given address")?;
-
-        log::info!("Development server at: http://{}", &address);
-
-        for mut stream in listener.incoming().filter_map(|x| x.ok()) {
-            respond_to_request(&mut stream, &build_dir_path).unwrap_or_else(|e| {
-                let _ = stream.write("HTTP/1.1 400 BAD REQUEST\r\n\r\n".as_bytes());
-                log::error!("an error occurred: {}", e);
-            });
-        }
+    pub fn start(mut self, command: Option<process::Command>) -> Result<()> {
```

Comment:
Last change: move this argument to a public field that also has builder pattern and we're good.

This is because the argument with an Option like that is not very explicit. Having a method for setting it makes it more explicit when you write and read the code later.